packages/data/src/types.ts

```diff
@@ -4,5 +4,24 @@
 export type WPDataSelectors = {
 	hasStartedResolution: ( selector: string, args?: string[] ) => boolean;
 	hasFinishedResolution: ( selector: string, args?: string[] ) => boolean;
-	isResolving: ( selector: string, args: [  ] ) => boolean;
+	isResolving: ( selector: string, args: string[] ) => boolean;
+};
+
+export type WPDataActions = {
+	startResolution: ( selector: string, args?: string[] ) => void;
+	finishResolution: ( selector: string, args?: string[] ) => void;
+};
+
+// Omitting state from selector parameter
+export type WPDataSelector< T > = T extends (
+	state: infer S,
+	...args: infer A
+) => infer R
+	? ( ...args: A ) => R
+	: T;
```

Comment:
See comment, we can use this for exporting selector types, this omits the first parameter (state), which is not necessary when calling the selector within a component.

Reply:
Yep great call on this one 👏